docker/usr/local/bin/download-and-extract

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Exit immediately on error
+set -e
+
+URL="$1"
+
+# Download the source code
+wget "${URL}"
+
+# Extract the source code archive
+ARCHIVE_FILE=$(find . -type f | head -n 1)
+case "${ARCHIVE_FILE}" in
+  *.tar.gz)  tar xf ${ARCHIVE_FILE} ;;
```

Comment:
`${ARCHIVE_FILE}` -> `"${ARCHIVE_FILE}"` just to make sure we don't get burnt by spaces (not that there should be any anyway)